### PR TITLE
Issue #159 Diffing a table with an empty list: Added two tests diffin…

### PR DESCRIPTION
**Pull request by mlvandijk**
…g with empty table to TableDifferTest

## Summary
Added two tests for issue #159

## Details
One test to verify that diffing a non-empty table with an empty table throws expected TableDiffException.
One test to verify that diffing an empty table with an empty list does not throw an Exception.

## Motivation and Context
Show that the issue has been resolved

## How Has This Been Tested?
Test code only, to show issue has been resolved.

## Types of changes
- [ x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:
- [x ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: core/src/test/java/cucumber/runtime/table/TableDifferTest.java [hunk 1]
```diff
@@ -229,6 +230,41 @@ public void should_diff_with_empty_list() {
     }
 
     @Test(expected = TableDiffException.class)
+    public void should_diff_with_empty_table() {
+        try {
+            DataTable emptyTable = DataTable.create(Collections.emptyList());
+            table().diff(emptyTable);
+        } catch (IndexOutOfBoundsException e) {
```
Comment (mpkorstanje) [inline core/src/test/java/cucumber/runtime/table/TableDifferTest.java]: You are expecting the test to throw a TableDiffException but catching an IndexOutOfBoundsException. I don't think this is quite right.

File: core/src/test/java/cucumber/runtime/table/TableDifferTest.java [hunk 2]
```diff
@@ -229,6 +230,41 @@ public void should_diff_with_empty_list() {
...
+        }
+    }
+
+    @Test
+    public void empty_list_should_not_diff_with_empty_table() {
+        try {
+            List<List<String>> emptyList = new ArrayList<List<String>>();
+            DataTable emptyTable = DataTable.create(Collections.emptyList());
+            emptyTable.diff(emptyList);
+        } catch (IndexOutOfBoundsException e) {
```
Comment (mpkorstanje) [inline core/src/test/java/cucumber/runtime/table/TableDifferTest.java]: In this test case we want to compare the empty table another another empty table. This should not result in any exception.

**Comment by mlvandijk** [comment]
@mpkorstanje Thanks for your feedback! Made some changes, hope it's ok now.

**Comment by olleolleolle** [comment]
@mlvandijk Hooray!

**Comment by mpkorstanje** [comment]
Thank you! 

I've squashed your commits and somewhat edited the commit message for brevity and clarity. You can read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) and [Why sqaush git commits](https://softwareengineering.stackexchange.com/questions/263164/why-squash-git-commits-for-pull-requests) for a more detailed explanation. We're not religious about it but it does improve communication.

One more thing to do: Update [History.md](/cucumber/cucumber-jvm/blob/master/History.md) with your contribution. If the commit-bit-bot is working you should have been granted commit access.

**Comment by mlvandijk** [comment]
@mpkorstanje Updated History.md (with shorter/cleared commit message I hope). Great links, thanks!

**Pull request merged**